### ISSUE TITLE
CS1929 is reported instead of CS1503 when extension methods share the same name.

### ISSUE THREAD

**Issue by otac0n** [dotnet/roslyn#83567]
**Version Used**: 
VS2026 Apr 2026, 18.5.2

**Steps to Reproduce**:

Minimal case:

<img width="1756" height="1519" alt="Image" src="https://github.com/user-attachments/assets/2cd30dd3-55cb-456b-b231-448771ea896f" />

```
using System.Numerics;

R<float> r = default;
r.GetScaled<float>(V<int>.Zero, V<float>.Zero); // This doesn't match any overload and should be an error.

struct V<T> where T : INumberBase<T>
{
    public static V<T> Zero = default;
}

struct B<T> where T : INumber<T>
{
}

struct R<T> where T : INumber<T>
{
}

static class B
{
    extension<TSelf>(B<TSelf> b)
        where TSelf : INumber<TSelf>
    {
        public B<TSelf> GetScaled(V<TSelf> scale, V<TSelf> anchor) => default;
    }
}

static class R
{
    extension<TSelf>(R<TSelf> r)
        where TSelf : INumber<TSelf>
    {
        public R<TSelf> GetScaled(V<TSelf> scale, V<TSelf> anchor) => default;
    }
}
```

**Diagnostic Id**:

CS1929 and CS1503

**Expected Behavior**:

The error message should consider the available overloads' receiver types before deciding what the "best" overload was.  The error should read:

`CS1503: Argument 2: cannot convert from 'V<int>' to 'V<float>'`

**Actual Behavior**:

In this case, an incorrect explanation of the error is given to the user.

`CS1929: 'R<float>' does not contain a definition for 'GetScaled' and the best extension method overload 'B.extension<float>(B<float>).GetScaled(V<float>, V<float>)' requires a receiver of type 'B<float>'`

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/roslyn/issues/33495
